The `tctl` derivation seems to only need:
```
__darwinAllowLocalNetworking = true;
```
but the `tctl-next` derivation fails because a test tries to access `/etc/protocols`:
```
running tests
ok      github.com/temporalio/tctl/cli  0.657s
ok      github.com/temporalio/tctl/cli/headers  0.012s
ok      github.com/temporalio/tctl/cli/stringify        0.016s
ok      github.com/temporalio/tctl/cli/trace    0.012s
panic: open /etc/protocols: operation not permitted

goroutine 1 [running]:
modernc.org/libc/honnef.co/go/netdb.init.0()
        /private/tmp/nix-build-tctl-next-0.9.0.drv-1/source/vendor/modernc.org/libc/honnef.co/go/netdb/netdb.go:700 +0x784
FAIL    github.com/temporalio/cli/app   0.043s
FAIL
```

--- a/pkgs/applications/networking/cluster/temporal-cli/darwin-sandbox-fix.patch
+++ b/pkgs/applications/networking/cluster/temporal-cli/darwin-sandbox-fix.patch
@@ -1,0 +1,20 @@
+--- vendor/modernc.org/libc/honnef.co/go/netdb/netdb.go
++++ vendor/modernc.org/libc/honnef.co/go/netdb/netdb.go
+@@ -696,7 +696,7 @@ func init() {
+ 	// Load protocols
+ 	data, err := ioutil.ReadFile("/etc/protocols")
+ 	if err != nil {
+-		if !os.IsNotExist(err) {
++		if !os.IsNotExist(err) && !os.IsPermission(err) {
+ 			panic(err)
+ 		}
+ 
+@@ -732,7 +732,7 @@ func init() {
+ 	// Load services
+ 	data, err = ioutil.ReadFile("/etc/services")
+ 	if err != nil {
+-		if !os.IsNotExist(err) {
++		if !os.IsNotExist(err) && !os.IsPermission(err) {
+ 			panic(err)
+ 		}
+ 

--- a/pkgs/applications/networking/cluster/temporal-cli/default.nix
+++ b/pkgs/applications/networking/cluster/temporal-cli/default.nix
@@ -1,6 +1,13 @@
 { lib, fetchFromGitHub, buildGoModule, installShellFiles, symlinkJoin }:
 
 let
+  overrideModAttrs = old: {
+    # https://gitlab.com/cznic/libc/-/merge_requests/10
+    postBuild = ''
+      patch -p0 < ${./darwin-sandbox-fix.patch}
+    '';
+  };
+
   tctl-next = buildGoModule rec {
     pname = "tctl-next";
     version = "0.9.0";
@@ -12,7 +19,9 @@ let
       hash = "sha256-zgi1wNx7fWf/iFGKaVffcXnC90vUz+mBT6HhCGdXMa0=";
     };
 
-    vendorHash = "sha256-muTNwK2Sb2+0df/6DtAzT14gwyuqa13jkG6eQaqhSKg=";
+    vendorHash = "sha256-EX1T3AygarJn4Zae2I8CHQrZakmbNF1OwE4YZFF+nKc=";
+
+    inherit overrideModAttrs;
 
     nativeBuildInputs = [ installShellFiles ];
 
@@ -33,6 +42,8 @@ let
         --bash <($out/bin/temporal completion bash) \
         --zsh <($out/bin/temporal completion zsh)
     '';
+
+    __darwinAllowLocalNetworking = true;
   };
 
   tctl = buildGoModule rec {
@@ -46,7 +57,9 @@ let
       hash = "sha256-LcBKkx3mcDOrGT6yJx98CSgxbwskqGPWqOzHWOu6cig=";
     };
 
-    vendorHash = "sha256-BUYEeC5zli++OxVFgECJGqJkbDwglLppSxgo+4AqOb0=";
+    vendorHash = "sha256-5wCIY95mJ6+FCln4yBu+fM4ZcsxBGcXkCvxjGzt0+dM=";
+
+    inherit overrideModAttrs;
 
     nativeBuildInputs = [ installShellFiles ];
 
@@ -63,6 +76,8 @@ let
         --bash <($out/bin/tctl completion bash) \
         --zsh <($out/bin/tctl completion zsh)
     '';
+
+    __darwinAllowLocalNetworking = true;
   };
 in
 symlinkJoin rec {

